californium-core/src/main/java/org/eclipse/californium/core/network/HashedMessageIdTracker.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations - initial creation
+ *    Achim Kraus (Bosch Software Innovations GmbH) - cleanup and use 
+ *                                                    System.nanoTime() instead
+ *                                                    of currentTimeMillis().
+ *    Achim Kraus (Bosch Software Innovations GmbH) - rename MessageIdTracker
+ *                                                    to HashedMessageIdTracker.
+ *                                                    introduce MessageIdTracker
+ *                                                    interface.
+ ******************************************************************************/
+package org.eclipse.californium.core.network;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.californium.core.coap.Message;
+
+/**
+ * A helper for keeping track of message IDs.
+ * <p>
+ * According to the
+ * <a href="https://tools.ietf.org/html/rfc7252#section-4.4">CoAP spec</a>
+ * 
+ * <pre>
+ * The same Message ID MUST NOT be reused (in communicating with the
+   same endpoint) within the EXCHANGE_LIFETIME (Section 4.8.2).
+ * </pre>
+ */
+public class HashedMessageIdTracker implements MessageIdTracker {
+
+	private final Map<Integer, Long> messageIds;
+	private final long exchangeLifetime; // milliseconds
```

Comment:
The millis are used by the previous implementation. I didn't spent too much time in normalize that, because I guess, this implementation will be removed in the future.